### PR TITLE
Fix deprecations with AGP 3.3

### PR DESCRIPTION
**Pull request by SUPERCILEX**
We have the `CompatibilityTest` to make sure things work on older versions.

### CODE REVIEW

File: plugin/src/main/kotlin/com/github/triplet/gradle/play/PlayPublisherPlugin.kt [hunk 1]
```diff
@@ -141,7 +141,12 @@ class PlayPublisherPlugin : Plugin<Project> {
                     "Processes packaging metadata for $variantName.",
                     null
             ) { init() }
-            checkManifest.dependsOn(processPackageMetadata)
+            try {
+                checkManifestProvider.configure { dependsOn(processPackageMetadata) }
+            } catch (e: NoSuchMethodError) {
+                @Suppress("DEPRECATION")
+                checkManifest.dependsOn(processPackageMetadata)
```
Comment (ChristianBecker) [inline plugin/src/main/kotlin/com/github/triplet/gradle/play/PlayPublisherPlugin.kt]: I fear we don't have another option besides relying on the `NoSuchMethodError` :(

File: plugin/build.gradle.kts [hunk 2]
```diff
@@ -9,11 +9,7 @@ plugins {
 }
 
 dependencies {
-    // Using compileOnly because every android project applies
-    // its own version of the Android Plugin.
-    // We're staying on 3.0.1 so we don't accidentally use APIs
-    // that are not available in the client project
-    compileOnly("com.android.tools.build:gradle:3.0.1")
+    compileOnly("com.android.tools.build:gradle:3.3.0-alpha11")
```
Comment (ChristianBecker) [inline plugin/build.gradle.kts]: `3.3.0-alpha12` was released yesterday. Or merge as-is and bump before release?

File: plugin/src/main/kotlin/com/github/triplet/gradle/play/PlayPublisherPlugin.kt (hunk 1 above)
Comment (SUPERCILEX) [reply plugin/src/main/kotlin/com/github/triplet/gradle/play/PlayPublisherPlugin.kt]: Yeah, I couldn't think of something else either.

File: plugin/build.gradle.kts (hunk 2 above)
Comment (SUPERCILEX) [reply plugin/build.gradle.kts]: I'll bump now

**Pull request merged**